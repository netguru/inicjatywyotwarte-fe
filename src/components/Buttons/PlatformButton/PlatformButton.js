import React from 'react'
import AndroidIcon from '@material-ui/icons/Android'
import AppleIcon from '@material-ui/icons/Apple'
import FacebookIcon from '@material-ui/icons/Facebook'
import styled from 'styled-components'

const Button = styled.a.attrs({
  'aria-label': 'odwiedź stronę z aplikacją',
  rel: 'noopener noreferrer',
  target: '_blank'
})`
  display: flex;
  align-items: center;
  justify-content: center;
  ${props =>
    props.size
      ? `
      height: ${props.size}px; 
      width: ${props.size}px;
      font-size: ${props.size / 1.7}px;`
      : `
      height: 2.5rem; 
      width: 2.5rem;
      font-size: 24px;
  `}

  color: ${props => props.theme.fontColorMedium};
  border: 1px solid ${props => props.theme.resourceTileLayoutColor};
  border-radius: 50%;
  cursor: pointer;
  transition: ${props => props.theme.buttonTransition};

  &:hover {
    background: ${props => props.theme.fontColorDark};
    color: ${props => props.theme.lightBackgroundColor};
    border-color: ${props => props.theme.fontColorDark};
  }
`
Button.displayName = 'PlatformButton';

export default function PlatformButton ({ url, platform, size }) {
  const icon =
    platform === 'android' ? (
      <AndroidIcon fontSize='inherit' />
    ) : platform === 'ios' ? (
      <AppleIcon fontSize='inherit' />
    ) : platform === 'fb' ? (
      <FacebookIcon fontSize='inherit' />
    ) : null

  return (
    <Button href={url} size={size}>
      {icon}
    </Button>
  )
}
