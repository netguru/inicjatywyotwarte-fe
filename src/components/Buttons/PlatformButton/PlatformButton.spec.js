import React from 'react'
import PlatformButton from './PlatformButton'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AndroidIcon from '@material-ui/icons/Android'
import AppleIcon from '@material-ui/icons/Apple'
import FacebookIcon from '@material-ui/icons/Facebook'

const baseProps = {
  url: 'http://example.test',
  platform: 'android',
  size: '50'
};

describe('PlatformButton', () => {
  test('renders button', () => {
    const wrapper = shallow(<PlatformButton {...baseProps} />);
    expect(wrapper.find('PlatformButton').length).toEqual(1);
  });

  test('renders proper size and url', () => {
    const wrapper = shallow(<PlatformButton {...baseProps} />);
    expect(wrapper.find('PlatformButton[href="http://example.test"][size="50"]'))
  });

  test('has proper icon', () => {
    const androidWrapper = shallow(<PlatformButton {...baseProps} />);
    expect(androidWrapper.children().first().equals(<AndroidIcon fontSize='inherit' />)).toEqual(true);

    const appleProps = {
      ...baseProps,
      platform: 'ios'
    }
    const appleWrapper = shallow(<PlatformButton {...appleProps} />);
    expect(appleWrapper.children().first().equals(<AppleIcon fontSize='inherit' />)).toEqual(true);

    const facebookProps = {
      ...baseProps,
      platform: 'fb'
    }
    const facebookWrapper = shallow(<PlatformButton {...facebookProps} />);
    expect(facebookWrapper.children().first().equals(<FacebookIcon fontSize='inherit' />)).toEqual(true);
  });

  test('match snapshot', () => {
    const wrapper = shallow(<PlatformButton {...baseProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


