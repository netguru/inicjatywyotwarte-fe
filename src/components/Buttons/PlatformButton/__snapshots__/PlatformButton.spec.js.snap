// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformButton match snapshot 1`] = `
<PlatformButton
  href="http://example.test"
  size="50"
>
  <AndroidIcon
    fontSize="inherit"
  />
</PlatformButton>
`;
